right/src/config_parser/parse_config.c

```diff
@@ -194,6 +213,41 @@ parser_error_t ParseConfig(config_buffer_t *buffer)
         MouseScrollState.baseSpeed = mouseScrollBaseSpeed;
         MouseScrollState.acceleratedSpeed = mouseScrollAcceleratedSpeed;
 
+        if (DataModelMajorVersion >= 7) {
+            MouseMoveState.axisSkew = mouseMoveAxisSkew;
+            MouseScrollState.axisSkew = mouseScrollAxisSkew;
+            DiagonalSpeedCompensation = diagonalSpeedCompensation;
+        }
+
+        // Update secondary roles
+
+        if (DataModelMajorVersion >= 7) {
+            SecondaryRoles_Strategy = secondaryRoles_Strategy;
+            SecondaryRoles_AdvancedStrategyDoubletapTime = secondaryRoles_AdvancedStrategyDoubletapTime;
+            SecondaryRoles_AdvancedStrategyTimeout = secondaryRoles_AdvancedStrategyTimeout;
+            SecondaryRoles_AdvancedStrategySafetyMargin = secondaryRoles_AdvancedStrategySafetyMargin;
+            SecondaryRoles_AdvancedStrategyTriggerByRelease = secondaryRoles_AdvancedStrategyTriggerByRelease;
+            SecondaryRoles_AdvancedStrategyDoubletapToPrimary = secondaryRoles_AdvancedStrategyDoubletapToPrimary;
+            switch (secondaryRoles_AdvancedStrategyTimeoutAction) {
+                case SerializedSecondaryRoleActionType_Primary:
+                    SecondaryRoles_AdvancedStrategyTimeoutAction = SecondaryRoleState_Primary;
+                    break;
+                case SerializedSecondaryRoleActionType_Secondary:
+                    SecondaryRoles_AdvancedStrategyTimeoutAction = SecondaryRoleState_Secondary;
+                    break;
+                default:
+                    return ParserError_InvalidSecondaryRoleActionType;
+            }
+        }
+
+        // Update misc
+
+        if (DataModelMajorVersion >= 7) {
+            DoubleTapSwitchLayerTimeout = doubletapTimeout;
+            DoubletapConditionTimeout = doubletapTimeout;
```

Comment:
@kareltucek I'm a bit lost. Please explain how the above two variables differ and why they're assigned the same value.

Reply:
Ah, sorry, forgotten about this.

Well, one controls layer switcher doubletaps. I.e., layer doubletap-to-lock.

The second one controls macro `ifDoubletap` timeout.

Here specifically they are assigned the same value because it is done the same way in set_command.c. 

Generally, as for the original reason, clearly I or we have decided that users will probably not want to control these separately, and therefore simplified the macro api by assigning them from a common configuration variable. 

As for why I keep them separate in the codebase, I don't see a strong reason not to, and probably feel better when each of the separate features has its own abstraction for its parameters.